### ISSUE TITLE
Phrase "Invalid Form Key. Please refresh the page." doesn't get translated

### ISSUE THREAD

**Issue by textarea** [magento/magento2#27889]
#24344 # Preconditions (*)
1. Magento composer 2.3.4 with a locale store view different from English.

### Steps to reproduce (*)
1. As a guest customer, go to login page, simulate expired session removing your browser cookies and then try to login (even with invalid credentials).

### Expected result (*)
1. Get the error phrase "Invalid Form Key. Please refresh the page." translated into the store view locale.

### Actual result (*)
1. Get the error phrase "Invalid Form Key. Please refresh the page." **in English**, always.

**Comment by textarea** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @textarea. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @textarea, here is your [Magento instance](https://i-27889-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27889-2-4-develop.instances.magento-community.engineering/admin_d34a](https://i-27889-2-4-develop.instances.magento-community.engineering/admin_d34a)
Login: `a7aca63e` Password: `f0082c42a24c`
Instance will be terminated in up to 3 hours.

**Comment by textarea** [comment]
The error message come from this point:
https://github.com/magento/magento2/blob/cd77da6b216c9aa33681dc2569ef1d5ad4e6a160/app/code/Magento/Customer/Controller/Account/LoginPost.php#L155

**Comment by engcom-Echo** [comment]
Hi, @textarea. Thank you for your report.

I think you can use a translation file for used language.
https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/translations/xlate.html

Please feel free to comment, reopen or create new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) 
if you are still facing this issue on the latest `2.4-develop` branch. Thank you for collaboration.

**Comment by textarea** [comment]
Hi @engcom-Echo,
why did you close this issue?
Were you unable to replicate it or have you not even tried?
Obviously I HAVE a translation file (Italian language) and it works for all the strings except the one of this issue.
I've found others italian magento 2 stores, completely translated, but not "Invalid Form Key....."

**Comment by engcom-Echo** [comment]
Hi, @textarea.
Unfortunately, I couldn't repeat an issue.

![image](https://user-images.githubusercontent.com/51681435/79840486-a3010780-83be-11ea-9aef-9a3a3fd859c1.png)
![image](https://user-images.githubusercontent.com/51681435/79840827-24f13080-83bf-11ea-8a69-1f9cc3083764.png)

Please provide advanced steps, I will try again. What language is used for the store by default?

**Comment by textarea** [comment]
Ok @engcom-Echo, thanks for reopening.
You are verifying the "The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later." error message, not the one "Invalid Form Key. Please refresh the page.".
So, please try these steps:
1 - go to login page
2 - after page loaded remove all xxx domain cookies (Chrome developer window "Application" -> "Storage" -> "Cookies" -> xxx domain)
3 - try to login, you will get the right error message

thanks

**Comment by engcom-Echo** [comment]
![image](https://user-images.githubusercontent.com/51681435/79843463-fa08db80-83c2-11ea-8036-af813d75cb7a.png)

**Comment by textarea** [comment]
Ok @engcom-Echo , you are viewing the translated message correctly.
Can I know what italian translation did you used?
Thanks for your test.

**Comment by engcom-Echo** [comment]
If your Default store is English, after remove all xxx domain cookies, you redirected to Default Store.
Now for Default Store, I used Italian.
Do you have two or more stores? I want to check this case.

**Comment by textarea** [comment]
Only one store with italian locale.

**Comment by textarea** [comment]
clean magento test, get english error message
http://mageita.textarea.it/customer/account/login

**Comment by engcom-Echo** [comment]
Make sure your language pack has a translation of the phrase "Invalid Form Key. Please refresh the page."

**Comment by textarea** [comment]
It is present

**Comment by textarea** [comment]
Which italian package did you used @engcom-Echo ?

**Comment by engcom-Echo** [comment]
Yes it is

**Comment by textarea** [comment]
Changed phrase in vendor/magento/module-customer/Controller/Account/LoginPost.php line 155, which below get translated, but the error message is still untranslated.

![Schermata 2020-04-21 alle 12 34 08](https://user-images.githubusercontent.com/1927403/79856464-adc29900-83cc-11ea-8a55-85186340a9c9.jpg)

**Comment by textarea** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @textarea. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @textarea, here is your [Magento instance](https://i-27889-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27889-2-4-develop.instances.magento-community.engineering/admin_540a](https://i-27889-2-4-develop.instances.magento-community.engineering/admin_540a)
Login: `013b2433` Password: `3d67fa44c6ee`
Instance will be terminated in up to 3 hours.

**Comment by mcleanvaimo** [comment]
We have this same issue and I believe I can see why it is happening. 

When the CSRF validator generates this message, the translation area has not yet been initialised. So the class `\Magento\Framework\Phrase` still only has the renderer `Magento\Framework\Phrase\Renderer\Placeholder` which doesn't apply translations. 

If you look in `\Magento\Framework\App\Area::_initTranslate` you can see that the renderer gets replaced with this: 

```
        \Magento\Framework\Phrase::setRenderer(
            $this->_objectManager->get(\Magento\Framework\Phrase\RendererInterface::class)
        );
```

And then translations can get applied correctly. But the CSRF exception has already been returned to the FE and this initialisation has never happened. 

I have a couple of quick patches that **do** fix this issue, but I'm not sure if it's a security choice to return this exception without first loading the design area. Can someone please advise if either of the attached patches are introducing more issues than they solve? 

[design-load.txt](https://github.com/magento/magento2/files/4747591/design-load.txt)
[translate-load.txt](https://github.com/magento/magento2/files/4747592/translate-load.txt)

**Comment by textarea** [comment]
Closing this issue because it has been fixed in 2.4.0 release
